mmv1/products/dataproc/Batch.yaml

```
@@ -0,0 +1,468 @@
+# Copyright 2023 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Batch'
+base_url: 'projects/{{project}}/locations/{{location}}/batches'
+self_link: 'projects/{{project}}/locations/{{location}}/batches/{{batch_id}}'
+create_url: 'projects/{{project}}/locations/{{location}}/batches?batchId={{batch_id}}'
+delete_url: 'projects/{{project}}/locations/{{location}}/batches/{{batch_id}}'
+collection_url_key: 'batches'
+immutable: true
+description: |
+  Dataproc Serverless Batches lets you run Spark workloads without requiring you to
+  provision and manage your own Dataproc cluster.
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 10
+      update_minutes: 5
+      delete_minutes: 5
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
```

Most of these fields are unused - see https://googlecloudplatform.github.io/magic-modules/develop/resource-reference/#async for more details.

```suggestion

timeouts: !ruby/object:Api::Timeouts
  insert_minutes: 10
  update_minutes: 5
  delete_minutes: 5
async: !ruby/object:Api::OpAsync
  operation: !ruby/object:Api::OpAsync::Operation
    base_url: '{{op_id}}'
```